writer.go

```diff
@@ -76,3 +75,12 @@ func open(paths []string) ([]zapcore.WriteSyncer, func(), error) {
 	}
 	return writers, close, errs.AsError()
 }
+
+// OpenWriteSyncers combines the passed set of WriteSyncer objects into a locked
+// WriteSyncer. It returns any error encountered.
+func OpenWriteSyncers(writers ...zapcore.WriteSyncer) (zapcore.WriteSyncer, error) {
+	if len(writers) == 0 {
+		return zapcore.AddSync(ioutil.Discard), nil
+	}
+	return zapcore.Lock(zapcore.NewMultiWriteSyncer(writers...)), nil
+}
```

Comment:
@akshayjshah the whole point of this PR was to get a function in the API that takes interfaces instead of strings as output file paths.

Please see my first comment